keps/sig-auth/3766-referential-authorization/README.md

```diff
@@ -82,59 +94,88 @@ been [used by
-cases other than Gateway API. We would like to move ReferenceGrant to a neutral home
-(ideally, under SIG Auth) in order to make it the canonical API for managing references
-across namespaces.
+Any project that wants to enable cross-namespace references currently has to
+choose between introducing a dependency on Gateway API's ReferenceGrant or
+creating a new API that would be partially redundant (leading to confusion for
+users).
+
+Recent interest between SIGs has made it clear that ReferenceGrant is wanted for
+use cases other than Gateway API. We would like to move ReferenceGrant to a
+neutral home (ideally, under SIG Auth) in order to make it the canonical API for
+managing references across namespaces.
+
+At the same time, Ingress and Gateway controllers are often deployed with read
+access to all Secrets in a cluster. This can be a significant security risk as
+these components often already sit on the edge of a cluster. Finding a way to
+compromise an Ingress or Gateway controller could potentially provide access to
+all Secrets in a cluster. Ideally, a solution that is related to referential
+authorization could also help mitigate this problem.
 
 ### Goals
 
-* Move ReferenceGrant to an API Group that SIG Auth manages
+* Introduce API(s) that enable the same capabilities that ReferenceGrant does
+  within Gateway API.
+* In addition to ReferenceGrant's capabilities, expand the scope of this KEP so
+  that it enables:
+  * Authorizing controllers to read only resources that have been referenced by
+    resources they are implementing.
+  * Building a more generic approach that will enable generic controllers and
+    authorizers to implement this API without prior knowledge of resources that
+    references are from or to.
 * Clearly define how ReferenceGrant should be used, including both current use
-  cases and guidance for future use cases
+  cases and guidance for future use cases.
 * Implement a library to ensure that ReferenceGrant is implemented consistently
-  by all controllers
+  by all controllers.
 
 ### Non-Goals
 
-* Develop an authorizer that will automatically implement ReferenceGrant for all
-  use cases. (It would be impossible to represent concepts like "all namespaces"
-  or label selectors that have become important for this KEP).
+* Advanced capabilities like label selectors and the ability to allow references
+  from any namespace are likely reasonable future extensions of this API but are
+  intentionally out of scope for the initial work.
 
 ## Proposal
 
-Move the existing ReferenceGrant resource into a new
-`authorization.k8s.io` API group, defined within the Kubernetes code base
-as part of the 1.27 release.
-
-We will take this opportunity to clarify and update the API after SIG Auth
-feedback. This resource will start with v1alpha1 as the API version.
-
+Create 3 new API types in a new `reference.authorization.k8s.io` API Group:
+
+1. **ClusterReferenceGrant:** This resource grants access to resources by
```

Comment:
I think it would be interesting to show the example where in kube today we implicitely allow Pods to reference Secrets/ConfigMaps/etc. within the same namespace with no extra authorization.

To get a clearer picture, it'd be quite informational to go through the k8s API and find all these references and document them. I guess in the future, kube itself could create default ClusterReferenceGrant objects which are deletable for users that don't want this behavior out of the box.

I think it can be confusing to users to find out that for some resources (Pods), they don't need a ClusterReferenceGrant to access the same-namespace Secret, but for some resources (HTTPRoute), they do.